### ISSUE TITLE
terminate called after throwing an instance of 'std::bad_alloc'   what():  std::bad_alloc

### ISSUE THREAD

**Issue by xzy-666** [tensorflow/tensorflow#49975]
I have  Cross-compile the  .a file and Cross-compile the label_image , but when I put  files  in the armv7, it is a error that _[./label_iamge:cannot execute binary file :Exec format error](url). I have check  the Cross-compile process ,it is seem that's ok.
Cross-compile tools :  g++-arm-linux-gnueabihf 、gcc-arm-linux-gnueabihf
arm: NXP i.MX6Q Cortex-A9 
root@EmbedSky-Board:/xzy/mix# lscpu
_Architecture:          armv7l
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    1
Core(s) per socket:    4
Socket(s):             1
Model name:            ARMv7 Processor rev 10 (v7l)
CPU max MHz:           996.0000
CPU min MHz:           792.0000
root@EmbedSky-Board:/xzy/mix# cat /proc/cpuinfo
processor       : 0 
model name      : ARMv7 Processor rev 10 (v7l)
BogoMIPS        : 6.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpd32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x2
CPU part        : 0xc09
CPU revision    : 10_

**Comment by tilakrayal** [comment]
@xzy-666 ,

In order to expedite the trouble-shooting process, could you please provide the following information
OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
TensorFlow installed from (source or binary):
TensorFlow version:
Python version:
Installed using virtualenv? pip? conda?:
Bazel version (if compiling from source):
GCC/Compiler version (if compiling from source):
CUDA/cuDNN version:
GPU model and memory:
 
and the exact sequence of commands / steps that you executed before running into the problem
Thanks!

**Comment by xzy-666** [comment]
hello， PC platform is linux ubuntu 16.04 ,python3.5.2 and armv7 is   Linux version 4.1.15  ,python3.5.2 . 
it is not  Mobile device that andriod or ios . it is arm-linux.
sorry, I not install TensorFlow, so , i just Cross-compile TF lite, tensorflow version 2.3.2

**Comment by tilakrayal** [comment]
@xzy-666 ,

Can you please provide the error you are facing which helps us to analyse the issue.
Also please take a look at this link for more information.[Link](https://www.tensorflow.org/lite/guide/build_arm).  Thanks!

**Comment by xzy-666** [comment]
@tilakrayal 
hello，I have use the command that _"./tensorflow/lite/tools/make/build_rpi_lib.sh ''_and _"./tensorflow/lite/tools/make/build_rpi_lib.sh ''_. I have get libtensorflow-lite.a and label_image.but, I put the files on the arm 。using the conmmand that _./label_image -v 4 -m ./mobilenet_v1_1.0_224.tflite -i ./grace_hopper.jpg    -l labels.txt_ ， I get the error .
_resolved reporter
tensors size: 88
nodes size: 31
inputs: 1
input(0) name: input
0: MobilenetV1/Conv2d_0/weights, 3456, 1, 0, 0
1: MobilenetV1/Conv2d_10_depthwise/depthwise_weights, 18432, 1, 0, 0
2: MobilenetV1/Conv2d_10_pointwise/weights, 1048576, 1, 0, 0
3: MobilenetV1/Conv2d_11_depthwise/depthwise_weights, 18432, 1, 0, 0
4: MobilenetV1/Conv2d_11_pointwise/weights, 1048576, 1, 0, 0
5: MobilenetV1/Conv2d_12_depthwise/depthwise_weights, 18432, 1, 0, 0
6: MobilenetV1/Conv2d_12_pointwise/weights, 2097152, 1, 0, 0
7: MobilenetV1/Conv2d_13_depthwise/depthwise_weights, 36864, 1, 0, 0
8: MobilenetV1/Conv2d_13_pointwise/weights, 4194304, 1, 0, 0
9: MobilenetV1/Conv2d_1_depthwise/depthwise_weights, 1152, 1, 0, 0
10: MobilenetV1/Conv2d_1_pointwise/weights, 8192, 1, 0, 0
11: MobilenetV1/Conv2d_2_depthwise/depthwise_weights, 2304, 1, 0, 0
12: MobilenetV1/Conv2d_2_pointwise/weights, 32768, 1, 0, 0
13: MobilenetV1/Conv2d_3_depthwise/depthwise_weights, 4608, 1, 0, 0
14: MobilenetV1/Conv2d_3_pointwise/weights, 65536, 1, 0, 0
15: MobilenetV1/Conv2d_4_depthwise/depthwise_weights, 4608, 1, 0, 0
16: MobilenetV1/Conv2d_4_pointwise/weights, 131072, 1, 0, 0
17: MobilenetV1/Conv2d_5_depthwise/depthwise_weights, 9216, 1, 0, 0
18: MobilenetV1/Conv2d_5_pointwise/weights, 262144, 1, 0, 0
19: MobilenetV1/Conv2d_6_depthwise/depthwise_weights, 9216, 1, 0, 0
20: MobilenetV1/Conv2d_6_pointwise/weights, 524288, 1, 0, 0
21: MobilenetV1/Conv2d_7_depthwise/depthwise_weights, 18432, 1, 0, 0
22: MobilenetV1/Conv2d_7_pointwise/weights, 1048576, 1, 0, 0
23: MobilenetV1/Conv2d_8_depthwise/depthwise_weights, 18432, 1, 0, 0
24: MobilenetV1/Conv2d_8_pointwise/weights, 1048576, 1, 0, 0
25: MobilenetV1/Conv2d_9_depthwise/depthwise_weights, 18432, 1, 0, 0
26: MobilenetV1/Conv2d_9_pointwise/weights, 1048576, 1, 0, 0
27: MobilenetV1/Logits/AvgPool_1a/AvgPool, 4096, 1, 0, 0
28: MobilenetV1/Logits/Conv2d_1c_1x1/BiasAdd, 4004, 1, 0, 0
29: MobilenetV1/Logits/Conv2d_1c_1x1/Conv2D_bias, 4004, 1, 0, 0
30: MobilenetV1/Logits/Conv2d_1c_1x1/weights, 4100096, 1, 0, 0
31: MobilenetV1/Logits/SpatialSqueeze, 4004, 1, 0, 0
32: MobilenetV1/MobilenetV1/Conv2d_0/Conv2D_bias, 128, 1, 0, 0
33: MobilenetV1/MobilenetV1/Conv2d_0/Relu6, 1605632, 1, 0, 0
34: MobilenetV1/MobilenetV1/Conv2d_10_depthwise/Relu6, 401408, 1, 0, 0
35: MobilenetV1/MobilenetV1/Conv2d_10_depthwise/depthwise_bias, 2048, 1, 0, 0
36: MobilenetV1/MobilenetV1/Conv2d_10_pointwise/Conv2D_bias, 2048, 1, 0, 0
37: MobilenetV1/MobilenetV1/Conv2d_10_pointwise/Relu6, 401408, 1, 0, 0
38: MobilenetV1/MobilenetV1/Conv2d_11_depthwise/Relu6, 401408, 1, 0, 0
39: MobilenetV1/MobilenetV1/Conv2d_11_depthwise/depthwise_bias, 2048, 1, 0, 0
40: MobilenetV1/MobilenetV1/Conv2d_11_pointwise/Conv2D_bias, 2048, 1, 0, 0
41: MobilenetV1/MobilenetV1/Conv2d_11_pointwise/Relu6, 401408, 1, 0, 0
42: MobilenetV1/MobilenetV1/Conv2d_12_depthwise/Relu6, 100352, 1, 0, 0
43: MobilenetV1/MobilenetV1/Conv2d_12_depthwise/depthwise_bias, 2048, 1, 0, 0
44: MobilenetV1/MobilenetV1/Conv2d_12_pointwise/Conv2D_bias, 4096, 1, 0, 0
45: MobilenetV1/MobilenetV1/Conv2d_12_pointwise/Relu6, 200704, 1, 0, 0
46: MobilenetV1/MobilenetV1/Conv2d_13_depthwise/Relu6, 200704, 1, 0, 0
47: MobilenetV1/MobilenetV1/Conv2d_13_depthwise/depthwise_bias, 4096, 1, 0, 0
48: MobilenetV1/MobilenetV1/Conv2d_13_pointwise/Conv2D_bias, 4096, 1, 0, 0
49: MobilenetV1/MobilenetV1/Conv2d_13_pointwise/Relu6, 200704, 1, 0, 0
50: MobilenetV1/MobilenetV1/Conv2d_1_depthwise/Relu6, 1605632, 1, 0, 0
51: MobilenetV1/MobilenetV1/Conv2d_1_depthwise/depthwise_bias, 128, 1, 0, 0
52: MobilenetV1/MobilenetV1/Conv2d_1_pointwise/Conv2D_bias, 256, 1, 0, 0
53: MobilenetV1/MobilenetV1/Conv2d_1_pointwise/Relu6, 3211264, 1, 0, 0
54: MobilenetV1/MobilenetV1/Conv2d_2_depthwise/Relu6, 802816, 1, 0, 0
55: MobilenetV1/MobilenetV1/Conv2d_2_depthwise/depthwise_bias, 256, 1, 0, 0
56: MobilenetV1/MobilenetV1/Conv2d_2_pointwise/Conv2D_bias, 512, 1, 0, 0
57: MobilenetV1/MobilenetV1/Conv2d_2_pointwise/Relu6, 1605632, 1, 0, 0
58: MobilenetV1/MobilenetV1/Conv2d_3_depthwise/Relu6, 1605632, 1, 0, 0
59: MobilenetV1/MobilenetV1/Conv2d_3_depthwise/depthwise_bias, 512, 1, 0, 0
60: MobilenetV1/MobilenetV1/Conv2d_3_pointwise/Conv2D_bias, 512, 1, 0, 0
61: MobilenetV1/MobilenetV1/Conv2d_3_pointwise/Relu6, 1605632, 1, 0, 0
62: MobilenetV1/MobilenetV1/Conv2d_4_depthwise/Relu6, 401408, 1, 0, 0
63: MobilenetV1/MobilenetV1/Conv2d_4_depthwise/depthwise_bias, 512, 1, 0, 0
64: MobilenetV1/MobilenetV1/Conv2d_4_pointwise/Conv2D_bias, 1024, 1, 0, 0
65: MobilenetV1/MobilenetV1/Conv2d_4_pointwise/Relu6, 802816, 1, 0, 0
66: MobilenetV1/MobilenetV1/Conv2d_5_depthwise/Relu6, 802816, 1, 0, 0
67: MobilenetV1/MobilenetV1/Conv2d_5_depthwise/depthwise_bias, 1024, 1, 0, 0
68: MobilenetV1/MobilenetV1/Conv2d_5_pointwise/Conv2D_bias, 1024, 1, 0, 0
69: MobilenetV1/MobilenetV1/Conv2d_5_pointwise/Relu6, 802816, 1, 0, 0
70: MobilenetV1/MobilenetV1/Conv2d_6_depthwise/Relu6, 200704, 1, 0, 0
71: MobilenetV1/MobilenetV1/Conv2d_6_depthwise/depthwise_bias, 1024, 1, 0, 0
72: MobilenetV1/MobilenetV1/Conv2d_6_pointwise/Conv2D_bias, 2048, 1, 0, 0
73: MobilenetV1/MobilenetV1/Conv2d_6_pointwise/Relu6, 401408, 1, 0, 0
74: MobilenetV1/MobilenetV1/Conv2d_7_depthwise/Relu6, 401408, 1, 0, 0
75: MobilenetV1/MobilenetV1/Conv2d_7_depthwise/depthwise_bias, 2048, 1, 0, 0
76: MobilenetV1/MobilenetV1/Conv2d_7_pointwise/Conv2D_bias, 2048, 1, 0, 0
77: MobilenetV1/MobilenetV1/Conv2d_7_pointwise/Relu6, 401408, 1, 0, 0
78: MobilenetV1/MobilenetV1/Conv2d_8_depthwise/Relu6, 401408, 1, 0, 0
79: MobilenetV1/MobilenetV1/Conv2d_8_depthwise/depthwise_bias, 2048, 1, 0, 0
80: MobilenetV1/MobilenetV1/Conv2d_8_pointwise/Conv2D_bias, 2048, 1, 0, 0
81: MobilenetV1/MobilenetV1/Conv2d_8_pointwise/Relu6, 401408, 1, 0, 0
82: MobilenetV1/MobilenetV1/Conv2d_9_depthwise/Relu6, 401408, 1, 0, 0
83: MobilenetV1/MobilenetV1/Conv2d_9_depthwise/depthwise_bias, 2048, 1, 0, 0
84: MobilenetV1/MobilenetV1/Conv2d_9_pointwise/Conv2D_bias, 2048, 1, 0, 0
85: MobilenetV1/MobilenetV1/Conv2d_9_pointwise/Relu6, 401408, 1, 0, 0
86: MobilenetV1/Predictions/Reshape_1, 4004, 1, 0, 0
87: input, 602112, 1, 0, 0
len: 61306
width, height, channels: -16842752, 1766213120, 246279780
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted_
I have looked up the data, and it seems that it is caused by insufficient memory. So I checked the memory of arm, and its _memory usage is as follows:
MemTotal:        2061132 kB
MemFree:         1794816 kB
MemAvailable:    1867312 kB
Buffers:            5924 kB
Cached:            87248 kB
SwapCached:            0 kB
Active:            68852 kB
Inactive:          34884 kB
Active(anon):      10688 kB
Inactive(anon):     9524 kB
Active(file):      58164 kB
Inactive(file):    25360 kB
Unevictable:           0 kB
Mlocked:               0 kB
HighTotal:        278528 kB
HighFree:         262148 kB
LowTotal:        1782604 kB
LowFree:         1532668 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Dirty:                 0 kB
Writeback:             0 kB
AnonPages:         10576 kB
Mapped:            15880 kB
Shmem:              9640 kB
Slab:              16204 kB
SReclaimable:       7048 kB
SUnreclaim:         9156 kB
KernelStack:         840 kB
PageTables:          396 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:     1030564 kB
Committed_AS:      67364 kB
VmallocTotal:     253952 kB
VmallocUsed:        9772 kB
VmallocChunk:     167732 kB
CmaTotal:         393216 kB
CmaFree:          257532 kB_
It took me a lot of time to solve this error, but I still couldn't find a way to solve it.

**Comment by advaitjain** [comment]
unassigning myself as this is a TfLite issue and I am focused on TfLite Micro.

**Comment by tilakrayal** [comment]
@xzy-666 ,

Can you please refer the above [comment](https://github.com/tensorflow/tensorflow/issues/49975#issuecomment-853905088) also please try to install the latest stable tf v2.6 and let us know if the issue still persists.Thanks
